internal/ethapi/api.go

```diff
@@ -1233,6 +1233,8 @@ func RPCMarshalBlock(block *types.Block, inclTx bool, fullTx bool) (map[string]i
 func (s *PublicBlockChainAPI) rpcMarshalHeader(ctx context.Context, header *types.Header) map[string]interface{} {
 	fields := RPCMarshalHeader(header)
 	fields["totalDifficulty"] = (*hexutil.Big)(s.b.GetTd(ctx, header.Hash()))
+	author, _ := s.b.Engine().Author(header)
```

Comment:
You can resolve the author field correctly in the `RPCMarshalHeader` function. It requires you to pass the consensus engine but it's available everywhere.

Reply:
`s.b.Engine().Author` need  `s *PublicBlockChainAPI`，but  `RPCMarshalBlock` do not have this.

Reply:
Maybe I can judge if Coinbase is empty, ecrecover extracts the Ethereum account address from a signed header.

Reply:
It's not good to me to do the recover operation in the API level. We should reuse the code in the consensus package.
You can pass the consensus.Engine into the `RPCMarshalBlock`. It's totally fine.

Reply:
thank you,I pass the consensus.Engine into the RPCMarshalBlock.